scripts/base/protocols/redis/main.zeek

```diff
@@ -139,10 +173,30 @@ event Redis::command(c: connection, cmd: Command)
 		}
 
 	++c$redis_state$current_command;
+
+	if ( cmd?$known )
+		{
+		if ( c$redis_state$resp_version == RESP2 )
+			{
+			local should_enter = cmd$known in enter_subscribed_mode;
+			local should_exit = cmd$known in exit_subscribed_mode;
+			c$redis_state$subscribed_mode = should_enter && ! should_exit;
+
+			# Warn if it's in both
+			if ( should_enter && should_exit )
+				Reporter::warning(fmt("Command '%s' enters then immediately exits subscribed mode",
```

Comment:
Maybe make this a "weird" instead?

Reply:
Is it better as a weird? Right now there should be no way to trigger since there just aren't commands that enter then exit, this will only realistically come up during development or if we allow redefing and someone adds a command in both enter and exit. Both of those I would classify more as a warning than a weird, but I'm not sure if other analyzers would go the weird route?

Reply:
If it's more of an `assert`, maybe you could use assert :man_shrugging: